components/content/src/utils.rs

```diff
@@ -58,6 +59,53 @@ pub fn find_related_assets(path: &Path, config: &Config, recursive: bool) -> Vec
     assets
 }
 
+/// Serializes assets source path for assets colocated with a section or a page
+pub fn serialize_assets(
+    assets: &Vec<PathBuf>,
+    parent_path: Option<&Path>,
+    colocated_path: Option<&String>,
+) -> Vec<String> {
+    assets
+        .iter()
+        .filter_map(|asset| asset.strip_prefix(parent_path.unwrap()).ok())
```

Comment:
Pass parent_path: &Path rather than the option if all we're doing is unwrapping it

Reply:
If `assets` is empty, I don't have any guarantee that `parent_path` will be Some.
I tried to reuse the same structure as the existing code but I can add an `if !assets.is_empty()` to unwrap this variable once instead of doing it in the iteration.
Doing it outside of `serialize_assets` to get rid of the `Option` in the signature would require to wrap the call of this function in the if statement mentionned above.
This would result in code duplication as I would have to do it twice, one for the pages and one for the sections.

What would you prefer me to implement?
1. Keep it as it is
2. Add the if statement in `serialize_assets` function to avoid unwrapping the option in the iteration
3. Add the if statement around the `serialize_assets`

Reply:
How much duplication are we talking about? I'm leaning towards 3

Reply:
We are just talking about a few lines (`if !assets.is_empty()` statement and of `parent_path` and `colocated_path` options unwrapping) duplicated twice, for pages and sections (where `serialize_assets` is currently used).

If we go for 3., we should apply the same logic to the `get_assets_permalinks` function by moving l.95-96 (`if !serialized_assets.is_empty()` statement and `colocated_path` option unwraping) around the function call.

Is it ok if I proceed that way?

Reply:
I think it's fine. Eventually page/sections will converge probably so it's going to reduce the duplication